code_sync/code_sync.py

```diff
@@ -2,39 +2,186 @@
+
+    local_dir = input('Path to code_sync on this local machine: ')
+    target = input('Destination machine: ')
+    remote_dir = input('Path on the destination machine to sync: ')
+    port = int(input('Port number to use (default 22): ') or "22")
+
+    config_entry_data = {
+        project: {
+            'local_dir': local_dir,
+            'target': target,
+            'remote_dir': remote_dir,
+            'port': port,
+
+        }
+    }
+
+    create_config_if_not_exists()
+    config_file_path = Path(Path.home(), CONFIG_FILE_NAME)
+    with open(config_file_path.__str__(), 'a') as f:
+        yaml.dump(config_entry_data, f, default_flow_style=False)
+
+    print("Successfully registered project '{}'".format(project))
+    return
+
+
+def list_projects() -> None:
+    """List all projects registered to code_sync."""
+    create_config_if_not_exists()
+    config = load_config()
+    if len(config) == 0:
+        print('No projects registered')
+    else:
+        formatted_keys = ', '.join(list(config.keys()))
+        print(formatted_keys)
+    return
+
+
+def identify_code_sync_parameters(args) -> Dict:
+    """
+    Identify the code_sync parameters. The user may specify a project (which should be registered to the code_sync
+        config) or specific all command line arguments.
+    Args:
+        args: The args object from argparse.
+
+    Returns:
+        Dictionary of the parameters to be used for the code_sync command.
+
+    Raises:
+        ValueError if the specified project is not registered to the code_sync config.
+    """
+    if args.project is not None:
+        config = load_config()
+        if args.project not in config:
+            raise ValueError("Project '{}' is not registered".format(args.project))
+        parameters = config[args.project]
+    else:
+        if args.local_dir is None or args.remote_dir is None or args.target is None:
+            raise ValueError('Missing argument. If a project is not specified, then local_dir, remote_dir, and target'
+                             ' must be specified.')
+        parameters = dict()
+        parameters['local_dir'] = args.local_dir
+        parameters['remote_dir'] = args.remote_dir
+        parameters['target'] = args.target
+        parameters['port'] = args.local_dir
+    return parameters
+
+
 def main():
     parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter, epilog=epilog_str)
-    parser.add_argument('--local_dir', help='the local code directory you want to sync', required=True)
-    parser.add_argument('--remote_dir', help='the remote directory you want to sync', required=True)
-    parser.add_argument('--target', help='specify which remote machine to connect to', required=True)
+    parser.add_argument('project', nargs='?', default=None)
+    parser.add_argument('--register', help='Register a new project to code_sync', required=False)
```

Comment:
The reason I did it the way I have it is so that you call `code_sync -- register <project name>` and that starts registering the `<project name>` project. Do you think it is a better interface to instead to `code_sync --register` and then code_sync asks for the name of the project?